src/shogun/base/SGObject.h

```
@@ -1026,6 +1033,20 @@ class CSGObject
 	/** Initialises all parameters with ParameterProperties::AUTO flag */
 	void init_auto_params();
 
+	/**
+	 * Set a default mask value that is added to any other mask
+	 * passed to watch_param. The default mask is by default
+	 * ParameterProperties::NONE.
+	 * @param mask the default mask
+	 */
+	void set_default_mask(ParameterProperties mask) noexcept
+	{
+		m_default_mask = mask;
+	}
+
+	/** The default mask for all registered parameters */
+	ParameterProperties m_default_mask = ParameterProperties::NONE;
```

i guess we could also avoid this member via applying the default mask to all yet registered parameters....downside would be that we need to always call this after registering parameters ....

---

yes, and for example in Features you want the behaviour in all derived classes, and this enforces it. Otherwise each subclass has to call such a method after registering all the parameters..

---

ah yes, very good point with the subclasses!